### PR TITLE
S6 canvas size

### PR DESCRIPTION
**Pull request by DanielGardiner**
- added index page with corresponding styles 
- added cookies JS file to set cookies
- added function to read cookie and generate canvas based on user input and user viewport

### CODE REVIEW

File: CSS/.idea/CSS.iml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (mporam) [inline CSS/.idea/CSS.iml]: please delete the idea folder from this file

File: JS/cookie.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+document.querySelector('#formSize').addEventListener('submit', function(e) {
+    e.preventDefault()
+    var value = document.querySelector('input[name="size"]:checked').value;
+    document.cookie = value;
```
Comment (mporam) [inline JS/cookie.js]: this is not how to set a cookie in js

File: JS/script.js [hunk 3]
```diff
@@ -34,19 +29,51 @@ function erase() {
     })
 }
 
+let canvasDimensions = calculateCanvasDimensions(document.cookie)
```
Comment (mporam) [inline JS/script.js]: accessing a cookie like this only works by chance

File: JS/script.js [hunk 4]
```diff
@@ -34,19 +29,51 @@ function erase() {
...
 
-function generateCanvas(row, column) {
+generateCanvas(canvasDimensions.width, canvasDimensions.height)
 
-    let canvasRow = '<div class="pixel"></div>'.repeat(column)
+function calculateCanvasDimensions(selectedCanvasSize) {
 
-    canvasRow = '<div class="row">' + canvasRow + '</div>'
+    let viewportWidth = Math.max(document.documentElement.clientWidth, window.innerWidth || 0)
```
Comment (mporam) [inline JS/script.js:51]: im suspicious that you know what this does? please comment an explanation

File: JS/script.js [hunk 5]
```diff
@@ -34,19 +29,51 @@ function erase() {
...
 
-    let canvas = canvasRow.repeat(row)
+    let viewportHeight = Math.max(document.documentElement.clientHeight, window.innerHeight || 0)
 
-    document.querySelector('.canvas').innerHTML = canvas
+    if (selectedCanvasSize === 'small') {
+
```
Comment (mporam) [inline JS/script.js]: why so much whitespace inside these ifs?

File: JS/script.js [hunk 6]
```diff
@@ -64,12 +91,13 @@ document.querySelector('.canvas').addEventListener('mouseleave', function() {
 })
 
 document.querySelector('.canvas').addEventListener('mouseenter', function() {
+
     if (clickdown === true) {
+
```
Comment (mporam) [inline JS/script.js]: why?

File: JS/script.js [hunk 7]
```diff
@@ -64,12 +91,13 @@ document.querySelector('.canvas').addEventListener('mouseleave', function() {
 })
 
 document.querySelector('.canvas').addEventListener('mouseenter', function() {
+
```
Comment (mporam) [inline JS/script.js]: why?

File: canvas.html [hunk 8]
```diff
@@ -29,10 +29,18 @@
...
+                </label>
+                <div class="tooltiptext">
+                    Start over
+                </div>
+            </button>
         </div>
     </nav>
     <div class="canvasContainer">
-        <div class="canvas"></div>
+            <div class="canvas"></div>
```
Comment (mporam) [inline canvas.html]: why?

**Review by mporam — CHANGES_REQUESTED** [review]

File: JS/cookie.js [hunk 9]
```diff
@@ -1,6 +1,4 @@
-document.querySelector('#formSize').addEventListener('submit', function(e) {
-    e.preventDefault()
-    var value = document.querySelector('input[name="size"]:checked').value;
-    document.cookie = value;
-    window.location.href = 'canvas.html'
+document.querySelector('#formSize').addEventListener('submit', function() {
+    let value = document.querySelector('input[name="size"]:checked').value;
+    document.cookie = "size="+value+"; expires=Thu, 18 Dec 2050 12:00:00 UTC";
```
Comment (mporam) [inline JS/cookie.js]: what happens after the 18th Dec 2050? the app will stop working!

File: canvas.html [hunk 10]
```diff
@@ -7,6 +7,7 @@
     <meta charset="UTF-8">
     <script defer src="JS/script.js"></script>
     <title>Cobra-Paint</title>
+    <script>console.log(undefined || 0)</script>
```
Comment (mporam) [inline canvas.html]: why have you added this in?

**Review by mporam — CHANGES_REQUESTED** [review]

File: canvas.html (hunk 10 above)
Comment (DanielGardiner) [reply canvas.html]: oops, experimenting

File: JS/cookie.js (hunk 9 above)
Comment (DanielGardiner) [reply JS/cookie.js]: have updated

File: canvas.html [hunk 11]
```diff
@@ -38,6 +37,10 @@
                     Start over
                 </div>
             </button>
+
```
Comment (mporam) [inline canvas.html]: no

**Review by mporam — CHANGES_REQUESTED** [review]

**Review by mporam — APPROVED** [review]

**Pull request merged**